Adds CRUD for Profiles

payne-chris-r opened the pull request:
Adds full feature for Profile resource. Attempts to keep some of the
user stuff out of the User model. Wondering if github name should
also be outside...
- Still getting a 500 error when you try to create multiple profiles,
which is better than it working, but still not an idea Rails error.

payne-chris-r committed:
Adds CRUD for Profiles

Adds full feature for Profile resource. Attempts to keep some of the
user stuff out of the User model. Wondering if github name should
also be outside...
- Still getting a 500 error when you try to create multiple profiles,
which is better than it working, but still not an idea Rails error.

sdavidson140 commented on scripts/profiles/create-profile.sh:
```diff
@@ -0,0 +1,19 @@
+  --header "Authorization: Token token=$TOKEN" \
+  --data '{
+    "profile": {
+      "user_id": "Practice",
```
remove sending `user_id` in requests. `user_id` is currently stored correctly based on current user

scottyscripts reviewed (approved):
Tested this with curl and rails console. There is still, issue #9 regarding 500 response on second profile creation, but will continue to update on issue.

scottyscripts committed:
profile curl scripts now using shell variables

- add script for get-profile
- delete profile script now correctly makes DELETE request

payne-chris-r commented:
@sdavidson140 ok to merge?
